src/fsharp/fsc.fs

```diff
@@ -2418,18 +2486,36 @@ module FSharpResidentCompiler =
                    None
             | None -> 
                 None
+#endif
 
-let main argv = 
-    let inline hasArgument name args = 
+#if FX_NO_EXIT
+let exit (_n:int) = failwith "System.Environment.Exit does not exist!"
```

Comment:
This is merely a placeholder for the API System.Environment.Exit, when the api is added back by the coreclr team, we will go back to using it.  The exit API, tears down the entire process.